Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,189 @@
+                    {
+                        EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamageShare));
+                        _popupSystem.PopupCoordinates(Loc.GetString("guardian-entity-taking-damage"), hostdamage.Owner.Transform.Coordinates, Filter.Entities(hostdamage.OwnerUid));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Triggers while trying to examine an activator to see if it's used
+        /// </summary>
+        private void OnActivatorExamined(EntityUid uid, GuardianCreatorComponent component, ExaminedEvent args)
+        {
+           if (component.Used == true)
+           {
+               string usedstring = Loc.GetString("guardian-activator-empty-invalid-creaton");
+               args.Message.AddMarkup("\n" + $"[color=#ba1919]{usedstring}[/color]");
+           }
+        }
+
+        /// <summary>
+        /// Called every time the host moves, to make sure the distance between the host and the guardian isn't too far
+        /// </summary>
+        private void OnGuardianHostMove(EntityUid uid, GuardianHostComponent component, ref MoveEvent args)
+        {
+            if (EntityManager.TryGetEntity(component.HostedGuardian, out IEntity? guard))
+            {
+                if  (guard.TryGetComponent<GuardianComponent>(out GuardianComponent? guardcomp))
+                {
+                    if (guardcomp.GuardianLoose == true)
+                    {
+                        //Compares the distance to allowed distance, otherwise forces a recall action from the host
+                        if (!guardcomp.Owner.Transform.Coordinates.InRange(EntityManager, component.Owner.Transform.Coordinates, guardcomp.DistanceAllowed))
+                        {
+                            OnGuardianManifestAction(guardcomp.OwnerUid, uid);
+                        }
+                    }
+                }        
+            }
+        }
+
+        /// <summary>
+        /// Called every time the guardian moves: makes sure it's not out of it's allowed distance
+        /// </summary>
+        private void OnGuardianMove(EntityUid uid, GuardianComponent component, ref MoveEvent args)
+        {
+            if (component.GuardianLoose == true)
+            {
+                //Compares the distance to allowed distance, otherwise forces a recall action from the host
+                if (!EntityManager.GetComponent<TransformComponent>(component.Host).Coordinates.InRange(EntityManager, EntityManager.GetComponent<TransformComponent>(uid).Coordinates, component.DistanceAllowed))
+                {
+                    OnGuardianManifestAction(uid, component.Host);
+                }
+            }
+        }
+
+        public void OnGuardianManifestAction(EntityUid guardian, EntityUid host)
+        {
+            var guardianloose = EntityManager.GetComponent<GuardianComponent>(guardian).GuardianLoose;
```

Comment:
Also don't use GetComponent unless you 100% know the component is there (in this case you don't).